### PR TITLE
Some cosmetics on the 3.8 changelogs

### PR DESCRIPTION
**Pull request by andreasabel**
Some cosmetics:
- capitalization, punctuation, formatting
- fixed some sentences
- moved `Cabal-syntax` to major changes section

- [ ] I wonder whether we will have a more detailed API change or migration guide for the `Cabal-syntax` decoupling.  

E.g., when upgrading `hackage-cli` to `Cabal-3.7` I noticed the following changes
- `readGenericPackageDescription` disappeared from `Distribution.PackageDescription`
- New module `Distribution.Simple.PackageDescription` (that exports `readGenericPackageDescription`)

This was easy enough to grep in the cabal sources, but I think new modules should be mentioned in the changelogs (the least).

- [ ] It seems that the changelog would benefit from more structuring into topics...

### CODE REVIEW

**Review by Mikolaj — APPROVED** [review]
Thank you. We will need to remember to use these fixes when re-generating changelogs for 3.8 proper (or for next RC, if any).

File: release-notes/cabal-install-3.8.0.20220526.md [hunk 1]
```diff
@@ -164,40 +163,68 @@ cabal-install 3.8.0.20220526 Changelog
   - Command-line `ghc-options` only applies to local packages
   - `program-options` stanza only applies to local packages
 
-- Improve error message for empty --allow-newer=  [#7740](https://github.com/haskell/cabal/issues/7740) [#8140](https://github.com/haskell/cabal/pull/8140)
+- Improve error message for empty `--allow-newer=`  [#7740](https://github.com/haskell/cabal/issues/7740) [#8140](https://github.com/haskell/cabal/pull/8140)
 
   Instead of internal error, the message now explains that empty argument for
-
+  `--allow-newer=` is not allowed and reminds what `--allow-newer` (with the empty
```
Comment (andreasabel) [inline release-notes/cabal-install-3.8.0.20220526.md:169]: This line was somehow lost when the changelog was assembled, I restored it from the PR.

File: release-notes/cabal-install-3.8.0.20220526.md [hunk 2]
```diff
@@ -164,40 +163,68 @@ cabal-install 3.8.0.20220526 Changelog
...
   - The error messages will provide a suggested fix if the misspelled extension or language is similar enough
     to known languages and extensions.
+
+### Internal changes
+
+- Remove bootstrapping plan files from version control, and simplify bootstrap update `Makefile` targets. [#7949](https://github.com/haskell/cabal/pull/7949)
+
+- CI setup [#6959](https://github.com/haskell/cabal/pull/6959)
+
+  - Remove travis scripts
```
Comment (ulysses4ever) [inline release-notes/cabal-install-3.8.0.20220526.md:230]:
```suggestion
- CI setup

  - Remove Travis scripts [#6959](https://github.com/haskell/cabal/pull/6959)
```

**Review by ulysses4ever — APPROVED** [review]
Looks much nicer, thank you!

**Comment by Mikolaj** [comment]
> Ah, ok, I simply assumed we would work with this `.md` file after initial generation from the changelog entries. But maybe you intended to regenerate the `.md` file?

I think regenerating the file is the easiest way if there are more than a couple new PRs or they are hard to identify.
 
> I assumed some manual editing had happened, because of the classification into 2 groups

Yes, unfortunately some editing is involved, especially that the tiny changelog files have not been deleted after a few previous versions and so we have a mix of old and new PRs.

> and because of https://github.com/haskell/cabal/pull/8214/files#r896092893.

When re-generating, let's make sure that was a human error, not a bug in the script.

**Comment by ulysses4ever** [comment]
What’s left for this to be merged?

**Comment by Mikolaj** [comment]
Nothing, Anybody can now add merge_me label and it's going to get merged. Usually the author does that, but authors are notoriously forgetful. :)

**Comment by Mikolaj** [comment]
@mergifyIO backport 3.8

**Pull request merged**